### DISCUSSION TITLE
Direnv not working

### DISCUSSION THREAD

**Discussion by jackinloadup** [srid/rust-nix-template#21] [Q&A] [answered]
Using `direnv` doesn't seem to start the shell but `nix develop` does.

```shell
direnv: loading ~/Projects/rust/rust-nix-template/.envrc
direnv: using flake
direnv: nix-direnv: using cached dev shell

### ️🔨 Welcome to the Nix devshell ###

Available commands:

## Commands

  , fmt    : Auto-format project tree
  , run    : Run the project executable
  , watch  : Watch for changes and run the project executable
/nix/store/09g3fqxa257s69yr080a989i2bham1zj-nix-direnv-2.3.0/share/nix-direnv/direnvrc:238: pop_var_context: head of shell_variables not a function context
/nix/store/09g3fqxa257s69yr080a989i2bham1zj-nix-direnv-2.3.0/share/nix-direnv/direnvrc:1073: pop_var_context: head of shell_variables not a function context
environment:1: pop_var_context: head of shell_variables not a function context
direnv: error exit status 1
```

I found [some information](https://discourse.nixos.org/t/bash-pop-var-context-head-of-shell-variables-not-a-function-context/26324) about how it stems from bash in nixpkgs-unstable but switching to 22.11 or 23.05 didn't help so unclear.

**Comment by srid** [comment]
@jackinloadup Isn't this an upstream nixpkgs issue? Is there something unique to rust-nix-template?

**Comment by jackinloadup** [comment]
@srid I'm not sure, the issue doesn't hinder using the template via `nix develop`. I just wanted to bring light to it in case others ran into the issue. If I figure it out I'll submit a patch. Feel free to close as you see fit.

**Comment by jackinloadup** [comment, marked as answer]
Pretty sure the root cause of this is resolved with https://github.com/Platonic-Systems/mission-control/pull/35. 

When direnv gave the following error `direnv: error exit status 1` it was taking issue the exit of the help menu in mission-control.
